pantheon-sessions.php

```diff
@@ -426,44 +620,77 @@ public function add_index() {
 		$query = "ALTER TABLE {$temp_clone_table} RENAME {$table};";
 		$wpdb->query( $query );
 
-		$this->safe_output( __( 'Operation complete, please verify that your site is working as expected. When ready, run wp pantheon session primary-key-finalize to clean up old data, or run wp pantheon session primary-key-revert if there were issues.', 'wp-native-php-sessions' ), 'log' );
+		return $output;
 	}
 
 	/**
 	 * Finalizes the creation of a primary key by deleting the old data.
+	 *
+	 * @param string $prefix Database prefix for the blog.
+	 * @param array $output An array of logs/errors which may occur.
+	 * @param bool $multisite Whether the site is a multisite.
 	 */
-	public function primary_key_finalize() {
+	public function primary_key_finalize_single( $prefix = null, $output = [], $multisite = false ) {
 		global $wpdb;
-		$table = $wpdb->prefix . 'bak_pantheon_sessions';
+		$table = esc_sql( $prefix . 'bak_pantheon_sessions' );
 
 		$query = $wpdb->prepare( 'SHOW TABLES LIKE %s', $wpdb->esc_like( $table ) );
 
 		// Check for table existence and delete if present.
 		if ( ! $wpdb->get_var( $query ) == $table ) {
-			$this->safe_output( __( 'Old table does not exist to be removed.', 'wp-native-php-sessions' ), 'error' );
+			// If dealing with multisites, it's feasible that some may have a
+			// table and some may not, so don't stop execution if it's not found.
+			if ( ! $multisite ) {
+				$type = 'error';
+			} else {
+				$output['no_old_table'] = $output['no_old_table'] + 1;
+				$type = 'log';
+			}
+
+			$this->safe_output( __( 'Old table does not exist to be removed.', 'wp-native-php-sessions' ), $type );
 		} else {
 			$query = "DROP TABLE {$table};";
 			$wpdb->query( $query );
 
-			$this->safe_output( __( 'Old table has been successfully removed, process complete.', 'wp-native-php-sessions' ), 'log' );
-
+			if ( ! $multisite ) {
+				$this->safe_output( __( 'Old table has been successfully removed.', 'wp-native-php-sessions' ), 'log' );
+			} else {
+				$this->safe_output( __( 'Old table has been successfully removed for this site.', 'wp-native-php-sessions' ), 'log' );
+			}
 		}
+
+		return $output;
 	}
 
 	/**
 	 * Reverts addition of primary key.
+	 *
+	 * @param string $prefix Database prefix for the blog.
+	 * @param array $output An array of logs/errors which may occur.
+	 * @param bool $multisite Whether the site is a multisite.
 	 */
-	public function primary_key_revert() {
+	public function primary_key_revert_single( $prefix = null, $output = [], $multisite = false ) {
 		global $wpdb;
-		$old_clone_table  = $wpdb->prefix . 'bak_pantheon_sessions';
-		$temp_clone_table = $wpdb->prefix . 'temp_pantheon_sessions';
-		$table            = $wpdb->prefix . 'pantheon_sessions';
+		$old_clone_table  = esc_sql( $prefix . 'bak_pantheon_sessions' );
+		$temp_clone_table = esc_sql( $prefix . 'temp_pantheon_sessions' );
+		$table            = esc_sql( $prefix . 'pantheon_sessions' );
```

Comment:
strings like these used in multiple methods should instead be class variables referenced. optional.

Reply:
They're used in multiple methods, but not in all methods, and they're never used by multiple methods in a single instantiation of the class. With that in mind, building them on every instance of the class seemed like unnecessary extra load.

Reply:
sorry, not the whole build, the literals `bak_pantheon_sessions`, `temp_pantheon_sessions`, etc.

Reply:
Ah... hrmm.... that's a good point and it's an easy tweak. One moment....